libs/ledger-live-common/src/families/bitcoin/wallet-btc/wallet.ts

```diff
@@ -8,67 +8,60 @@ import { TransactionInfo, DerivationModes } from "./types";
 import { Account, SerializedAccount } from "./account";
 import Xpub from "./xpub";
 import { IExplorer } from "./explorer/types";
-import BitcoinLikeExplorer from "./explorer";
 import { IStorage, Output } from "./storage/types";
 import BitcoinLikeStorage from "./storage";
 import { PickingStrategy } from "./pickingstrategies/types";
 import * as utils from "./utils";
 import cryptoFactory from "./crypto/factory";
+import BitcoinLikeExplorer from "./explorer";
 import { TX, Address } from "./storage/types";
+import { blockchainBaseURL } from "../../../api/Ledger";
+import { getCryptoCurrencyById } from "@ledgerhq/cryptoassets";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
 
 class BitcoinLikeWallet {
-  explorerInstances: { [key: string]: IExplorer } = {};
+  explorers: { [currencyId: string]: IExplorer } = {};
 
-  explorers: { [key: string]: (explorerURI: string) => IExplorer } = {
-    ledgerv3: (explorerURI) =>
-      new BitcoinLikeExplorer({
-        explorerURI,
-        explorerVersion: "v3",
-      }),
-    ledgerv2: (explorerURI) =>
-      new BitcoinLikeExplorer({
-        explorerURI,
-        explorerVersion: "v2",
-      }),
-  };
+  constructor() {}
 
   accountStorages: { [key: string]: (...args: any[]) => IStorage } = {
     mock: () => new BitcoinLikeStorage(),
   };
 
-  getExplorer(
-    explorer: "ledgerv3" | "ledgerv2",
-    explorerURI: string
-  ): IExplorer {
-    const id = `explorer-${explorer}-uri-${explorerURI}`;
-    this.explorerInstances[id] =
-      this.explorerInstances[id] || this.explorers[explorer](explorerURI);
-    return this.explorerInstances[id];
+  getExplorer(currency: CryptoCurrency) {
+    if (!this.explorers[currency.id]) {
+      this.explorers[currency.id] = new BitcoinLikeExplorer({
+        cryptoCurrency: currency,
+      });
+    }
+    return this.explorers[currency.id];
   }
 
-  async generateAccount(params: {
-    xpub: string;
-    path: string;
-    index: number;
-    currency: Currency;
-    network: "mainnet" | "testnet";
-    derivationMode: DerivationModes;
-    explorer: "ledgerv3" | "ledgerv2";
-    explorerURI: string;
-    storage: "mock";
-    storageParams: any[];
-  }): Promise<Account> {
+  async generateAccount(
+    params: {
+      xpub: string;
+      path: string;
+      index: number;
+      // FIXME: currency should be removed and instead use CryptoCurrency
+      currency: Currency;
+      network: "mainnet" | "testnet";
+      derivationMode: DerivationModes;
+      storage: "mock";
+      storageParams: any[];
```

Comment:
not related to the PR but any way to type this ?

Reply:
I have no idea, but it looks like btc related, do you know what it stores @hzheng-ledger ?

Reply:
Done in pair with huiqi